ecs-cli/modules/cli/compose/containerconfig/containerconfig.go

```diff
@@ -1,20 +1,32 @@
 package containerconfig
 
+import "github.com/aws/aws-sdk-go/service/ecs"
+
 // ContainerConfig all compose fields supported by the ecs-cli
 // TODO: finalize fields
 type ContainerConfig struct {
 	Name string
 
-	CapAdd      []string
-	CapDrop     []string
-	Command     string // type TBD
-	Devices     []string
-	DNS         []string // StringList in docker/cli
-	Entrypoint  string   // ShellCommand in docker/cli
-	Environment []string // MappingWithEquals in docker/cli
-	EnvFile     []string // StringList in docker/cli
-	ExtraHosts  []string
-	Hostname    string
-	HealthCheck interface{} // *HealthCheckConfig in docker/cli
-	Image       string
+	CapAdd                []string
+	CapDrop               []string
+	Command               []string
+	Devices               []string
+	DNSSearchDomains      []string
+	DNSServers            []string
+	DockerLabels          map[string]*string
+	DockerSecurityOptions []string
+	Entrypoint            []string
+	Environment           []*ecs.KeyValuePair
+	ExtraHosts            []string
+	Hostname              string
+	HealthCheck           *ecs.HealthCheck
+	Image                 string
+	Links                 []string
+	LoggingConfiguration  *ecs.LogConfiguration
+	PortMappings          []*ecs.PortMapping
+	Privileged            bool
+	ReadOnly              bool
+	Tmpfs                 []string
```

Comment:
Assuming that this `ContainerConfig` is as close to the `ecs.ContainerDefinition` as defined by the `aws-sdk-go`, is there a reason why this is at the top level for `ContainerConfig` (versus wrapped in LinuxParameters`)? Also, should `shmSize`/`SharedMemorySize` be included somewhere?

Reply:
like CapAdd, CapDrop, and Devices, this requires some processing into AWS types before adding to the task def:
https://github.com/allisaurus/amazon-ecs-cli/blob/dev/ecs-cli/modules/utils/compose/convert_task_definition.go#L687

My thought was we'd leave this kind of conversion in one spot (task def conversion) so it can be leveraged for v1/2 and v3 projects. However, we can also pull that logic into a separate util func that both parsing funcs can use. Thoughts?

Reply:
Oh I see. I misunderstood this type as being more aligned with the libcompose structure (so that presumably the conversion logic would remain more or less the same).  In that case, I'd just say that `ShmSize` should be in here at the same level as `Tmpfs`.